### DISCUSSION TITLE
PTScotch Compilation issue on HPC

### DISCUSSION THREAD

**Discussion by ajsummers** [idaholab/moose#16711] [Q&A Getting Started] [answered]
Let me know if I didn't provide enough information about my environment. I'm trying to install MOOSE on an HPC cluster (I don't have root privileges).

I started off using the standard Linux installation method. Installed Miniconda and the Moose environment, then cloned in the repository. After this, I followed the GCC/MPICH manual installation method.

OS: CentOS 6.10
Instead of installing gcc, mpich, and cmake manually, I added the modules from the HPC's library:
```
module load gcc/8.1.0
module load mpich/3.2.1

export PACKAGES_DIR=/tools

export PATH=$PACKAGES_DIR/gcc-8.1.0/bin:$PACKAGES_DIR/mpich-3.2.1/bin:$PATH
export LD_LIBRARY_PATH=$PACKAGES_DIR/gcc-8.1.0/lib64:$PACKAGES_DIR/gcc-8.1.0/lib:$PACKAGE$
export C_INCLUDE_PATH=$PACKAGES_DIR/mpich-3.2.1/include:$C_INCLUDE_PATH
export CPLUS_INCLUDE_PATH=$PACKAGES_DIR/mpich-3.2.1/include:$CPLUS_INCLUDE_PATH
export FPATH=$PACKAGES_DIR/mpich-3.2.1/include:$FPATH
export MANPATH=$PACKAGES_DIR/mpich-3.2.1/share/man:$MANPATH
export CC=mpicc
export CXX=mpicxx
export FC=mpif90
export F90=mpif90

module load cmake/gcc/3.12.3
```
When running
**./scripts/update_and_rebuild_petsc.sh --download-mumps=0 --with-64-bit-indices=1**

I get the following error:
![image](https://user-images.githubusercontent.com/64990120/104772688-6db0fc80-5739-11eb-847e-61a6b7b85c4b.png)

I've attached the configure.log file in the petsc directory below.

I read a post somewhere that hinted at a compatibility issue between MUMPS and PTScotch, but I'm not sure.

Thank you in advance for any help!
[configure.log](https://github.com/idaholab/moose/files/5822695/configure.log)

**Comment by mangerij** [comment]
I saw this earlier today.

You can disable the PTScotch flag to get around it if possible

On Jan 15, 2021 10:10 PM, ajsummers <notifications@github.com> wrote:

*Message sent from a system outside of UConn.*


Let me know if I didn't provide enough information about my environment. I'm trying to install MOOSE on an HPC cluster (I don't have root privileges).

I started off using the standard Linux installation method. Installed Miniconda and the Moose environment, then cloned in the repository. After this, I followed the GCC/MPICH manual installation method.

OS: CentOS 6.10
Instead of installing gcc, mpich, and cmake manually, I added the modules from the HPC's library:

module load gcc/8.1.0
module load mpich/3.2.1

export PACKAGES_DIR=/tools

export PATH=$PACKAGES_DIR/gcc-8.1.0/bin:$PACKAGES_DIR/mpich-3.2.1/bin:$PATH
export LD_LIBRARY_PATH=$PACKAGES_DIR/gcc-8.1.0/lib64:$PACKAGES_DIR/gcc-8.1.0/lib:$PACKAGE$
export C_INCLUDE_PATH=$PACKAGES_DIR/mpich-3.2.1/include:$C_INCLUDE_PATH
export CPLUS_INCLUDE_PATH=$PACKAGES_DIR/mpich-3.2.1/include:$CPLUS_INCLUDE_PATH
export FPATH=$PACKAGES_DIR/mpich-3.2.1/include:$FPATH
export MANPATH=$PACKAGES_DIR/mpich-3.2.1/share/man:$MANPATH
export CC=mpicc
export CXX=mpicxx
export FC=mpif90
export F90=mpif90

module load cmake/gcc/3.12.3


When running
./scripts/update_and_rebuild_petsc.sh --download-mumps=0 --with-64-bit-indices=1

I get the following error:
[image]<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F64990120%2F104772688-6db0fc80-5739-11eb-847e-61a6b7b85c4b.png&data=04%7C01%7Cjohn.mangeri%40uconn.edu%7C0c403f25836e43026b3e08d8b9918783%7C17f1a87e2a254eaab9df9d439034b080%7C0%7C0%7C637463382030424332%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=dlsW6dV%2Ffco4NRhNrCjWPa3Wok9cj%2BbrludoSfmFwKY%3D&reserved=0>

I've attached the configure.log file in the petsc directory below.

I read a post somewhere that hinted at a compatibility issue between MUMPS and PTScotch, but I'm not sure.

Thank you in advance for any help!
configure.log<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fidaholab%2Fmoose%2Ffiles%2F5822695%2Fconfigure.log&data=04%7C01%7Cjohn.mangeri%40uconn.edu%7C0c403f25836e43026b3e08d8b9918783%7C17f1a87e2a254eaab9df9d439034b080%7C0%7C0%7C637463382030434326%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Ds9fRGe%2BGj00fEpJQ3hDoQckRx4F%2F5ZgYckxMC52sCc%3D&reserved=0>

**Reply by ajsummers** [reply]
Thank you for your response.

I tried including the following flags on the script ./scripts/update_and_rebuild_petsc.sh:
--download-ptscotch=0
--with-ptscotch=0

This seemed to have teased out a potential problem with my C++ compiler. I'm going to check more into this.

Would it be advisable to continue in the installation without ptscotch?

**Reply by fdkong** [reply]
The simulation will be fine even without ptscotch. Most of partitions are handled by parmetis or Metis

**Comment by fdkong** [comment]
1) You might send us a wrong configure.log since it did not say anything about ptscoth. 

2) You might not need `conda` on HPC. What I would like to do is to module load gcc, mpich and cmake, and then build PETSc/libmesh/moose.

**Reply by ajsummers** [reply]
On the first note: I just noticed this. I assumed that the configure.log file directly under the petsc directory would be the correct one. Is there somewhere else I should be looking?

I will try installing moose without the conda environment and update this discussion soon.

**Reply by fdkong** [reply]
> On the first note: I just noticed this. I assumed that the configure.log file directly under the petsc directory would be the correct one. Is there somewhere else I should be looking?

It is the right place. But I do not know why you got an incomplete configure.log. The right one might be just overwritten when you tried again

**Reply by ajsummers** [reply]
After disabling ptscotch briefly, I found that there was a problem with my C compiler. I added mpc/1.0.3 and this seemed to fix the issues as far as installing MOOSE. I tried the update_and_rebuild_petsc.sh and it worked. I was disappointed in my system's default Python 2, so I went back and tried the standard linux conda install method, and that also worked.

I am now having similar problems with compiling my first project, but I will probably start a new thread for this. How should I designate the answer for this thread?

**Reply by fdkong** [reply]
<img width="213" alt="Screen Shot 2021-01-22 at 2 33 22 PM" src="https://user-images.githubusercontent.com/10508374/105550608-da3e7500-5cbe-11eb-91d5-595aa9c90b23.png">

**Reply by ajsummers** [reply]
I meant should I label yours as the answer even though the problem/solution ended up being different. Just so I can close it out as being "solved"

**Reply by GiudGiud** [reply]
It's better to re-post your answer as an answer and select it there. nbd here because the thread is so short anyway, so it's as you prefer.

**Comment by ajsummers** [comment, marked as answer]
After running the compilation without ptscotch, errors indicated that there might be an issue with my C compiler. Particularly a reference to a 'libmpc.so' file. I loaded a module 'mpc' from the HPC, and this fixed the issue. I was then able to install moose through conda.
